packages/shared/src/utils/templating.ts

```diff
@@ -5,15 +5,85 @@ export interface ITemplatedStringVariable {
   variables?: { [key: string]: ITemplatedStringVariable };
 }
 
+type ITemplatedDataContext = { [prefix: string]: any };
+
 /**
+ * Utility class for parsing data containing templated values
  *
- * @param value
- * @param context
+ * @param context - json object specifying specific values for replacement
+ * E.g. {row:{id:'example_1'}} will replace `@row.id` with 'example_1`
+ */
+export class TemplatedData {
```

Comment:
New generic class for taking any data which may or may not contain templated data in some form (e.g. @row statements), and provide methods to parse